src/Core__Option.res

```diff
@@ -22,73 +22,83 @@
  * along with this program; if not, write to the Free Software
  * Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA 02111-1307, USA. */
 
+let flat = opt =>
+  switch opt {
+  | Some(v) => v
+  | None => None
+  }
+
 let filterU = (opt, p) =>
   switch opt {
-  | Some(x) as some if p(. x) => some
+  | Some(value) as some if p(. value) => some
   | _ => None
   }
 
-let filter = (opt, p) => filterU(opt, (. x) => p(x))
+let filter = (opt, p) => filterU(opt, (. value) => p(value))
 
 let forEachU = (opt, f) =>
   switch opt {
-  | Some(x) => f(. x)
+  | Some(value) => f(. value)
   | None => ()
   }
 
-let forEach = (opt, f) => forEachU(opt, (. x) => f(x))
+let forEach = (opt, f) => forEachU(opt, (. value) => f(value))
 
-let getExn = x =>
-  switch x {
-  | Some(x) => x
+let getExn = opt =>
+  switch opt {
+  | Some(value) => value
   | None => raise(Not_found)
   }
 
+let expect = (opt, message) =>
+  switch opt {
+  | Some(value) => value
+  | None => raise(Failure(message))
```

Comment:
Another thought @glennsl - should this use the new fine `Error.panic`? There's a separate PR for incorporating `panic` in existing APIs, but since this isn't pre-existing I'm thinking we can add it directly here if we want.

Reply:
Done in https://github.com/rescript-association/rescript-core/pull/57/commits/47b16c738b9cacd8a57e02c2aa760f4c5de5b064